de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java

```diff
@@ -110,20 +110,23 @@ private SaveMetadataCallback(SaveAsDialog save_dialog) {
         }
 
         @Override
-         public void onFailure(Throwable caught) {
-             save_dialog.hide();
-             announcer.schedule(new ErrorAnnouncementConfig("Unable to save your file. Please try again or contact support."));
-         }
+        public void onFailure(Throwable caught) {
+            save_dialog.unmask();
+            if (caught.getMessage().contains("ERR_EXISTS")) {
+                announcer.schedule(new ErrorAnnouncementConfig(appearance.fileExistsError()));
+            } else {
+                ErrorHandler.post("Unable to save your file. Please try again or contact support.",
+                                  caught);
```

Comment:
And I don't know why I spaced and left this string out of the appearance, I'll move that over too.